docs/compilers/osv.md

```diff
@@ -16,3 +29,161 @@ Compiling Java on the OSv platform requires the following parameters be met:
+file `run.yaml` in it. In this file you specify where your entrypoint file is. For example, if your
+entrypoint file is `server.js`, then the final content of `meta/run.yaml` would be:
+```
+config_set:
+   conf1:
+      main: /server.js
+config_set_default: conf1
+```
+Optionally, you can create file `manifest.yaml` directly inside your project root directory
+where you specify some additional parameters. Parameter `image_size` specifys filesystem size
+of the created unikenrel.
+
+### Build command
+Use this command to build unikernel with your NodeJS application:
+```
+$ cd $PROJECT_ROOT
+$ unik build
+   --name myImg
+   --path ./
+   --base osv
+   --language nodejs
+   --provider openstack
+```
+
+## Native
+Native runtime can be used either for running your C/C++ application or for running
+arbitrary precompiled application that exists in public repository (these are called
+application packages).
+
+### Running application package
+Consult built-in documentation to see what configuration files are needed by UniK:
+```
+$ unik describe-compiler
+   --base osv
+   --language native
+   --provider openstack
+
+OVERVIEW
+Language "native" allows you to either:
+a) run precompiled package from the remote repository
+b) run your own binary code
+c) combination of (a) and (b)
+In case you intend to use packages from the remote repository, please provide
+meta/run.yaml where you list desired packages.
+In any case (i.e. (a), (b), (c)) you need to provide meta/run.yaml. See below for
+more details.
+
+HOW TO PREPARE APPLICATION
+(this is only needed if you want to run your own C/C++ application)
+Compile your application into relocatable shared-object (a file normally
+given a ".so" extension) that is PIC (position independent code).
+
+CONFIGURATION FILES
+------ /manifest.yaml ------
+image_size: "10GB"  # logical image size
+
+------ /meta/run.yaml ------
+config_set:
+   conf1:
+      bootcmd: <boot-command-that-starts-application>
+config_set_default: conf1
+
+------ /meta/package.yaml ------
+title: <your-unikernel-title>
+name: <your-unikernel-name>
+author: <your-name>
+require:
+  - <first-required-package-title>
+  - <second-required-package-title>
+  # ...
+
+OTHER
+Below please find a list of packages in remote repository:
+
+eu.mikelangelo-project.app.hadoop-hdfs
+eu.mikelangelo-project.app.mysql-5.6.21
+eu.mikelangelo-project.erlang
+eu.mikelangelo-project.ompi
+eu.mikelangelo-project.openfoam.core
+eu.mikelangelo-project.openfoam.pimplefoam
+eu.mikelangelo-project.openfoam.pisofoam
+eu.mikelangelo-project.openfoam.poroussimplefoam
+eu.mikelangelo-project.openfoam.potentialfoam
+eu.mikelangelo-project.openfoam.rhoporoussimplefoam
+eu.mikelangelo-project.openfoam.rhosimplefoam
+eu.mikelangelo-project.openfoam.simplefoam
+eu.mikelangelo-project.osv.cli
+eu.mikelangelo-project.osv.cloud-init
+eu.mikelangelo-project.osv.httpserver
+eu.mikelangelo-project.osv.nfs
```

Comment:
Even though I don't like very much, I hope we can pass for the first iteration. The list of available packages could be retrieved via `capstan package search` command, but since we don't want to mess around too much with Capstan we provide the list here. However, I think we can provide an external script that will always list the current set of packages and then provide the link here.